pkg/logs/docker/docker.go

```diff
@@ -28,18 +30,76 @@ func ParseMessage(msg []byte) (string, string, []byte, error) {
 		return "", "", nil, errors.New("Can't parse docker message: expected a 8 bytes header")
 	}
 
+	preMessageLength := getPreMessageLength(msg)
+	if len(msg[preMessageLength:]) > maxDockerBuffer {
+		msg = removePartialHeaders(msg, preMessageLength)
+	}
+
 	// First byte is 1 for stdout and 2 for stderr
 	status := message.StatusInfo
 	if msg[0] == 2 {
 		status = message.StatusError
 	}
 
+	// log.Errorf("msg[0] is %s or %c or %x or %U", msg[0], msg[0], msg[0], msg[0])
+	// log.Errorf("msg[1] is %s or %c or %x or %U", msg[1], msg[1], msg[1], msg[1])
+	// log.Errorf("msg[2] is %s or %c or %x or %U", msg[2], msg[2], msg[2], msg[2])
+	// log.Errorf("msg[3] is %s or %c or %x or %U", msg[3], msg[3], msg[3], msg[3])
+	// log.Errorf("msg[4] is %s or %c or %x or %U", msg[4], msg[4], msg[4], msg[4])
+	// log.Errorf("msg[5] is %s or %c or %x or %U", msg[5], msg[5], msg[5], msg[5])
+	// log.Errorf("msg[6] is %s or %c or %x or %U", msg[6], msg[6], msg[6], msg[6])
+	// log.Errorf("msg[7] is %s or %c or %x or %U", msg[7], msg[7], msg[7], msg[7])
+	// log.Errorf("msg[8] is %s or %c or %x or %U", msg[8], msg[8], msg[8], msg[8])
+
 	// timestamp goes from byte 8 till first space
 	to := bytes.Index(msg[messageHeaderLength:], []byte{' '})
 	if to == -1 {
 		return "", "", nil, errors.New("Can't parse docker message: expected a whitespace after header")
 	}
 	to += messageHeaderLength
 	ts := string(msg[messageHeaderLength:to])
+
+	// log.Errorf("msg[to+1:] is %s", msg[to+1:])
+	// log.Errorf("new length of message is %d", len(msg[to+1:]))
+
 	return ts, status, msg[to+1:], nil
+
+}
+
+// get length of the 8 byte header, timestamp, and space
+func getPreMessageLength(msg []byte) int {
+	idx := bytes.Index(msg, []byte{' '})
+	if idx == -1 {
+		return 0
+	}
+	return idx + 1
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// function only used if message is longer than 16kb
+func removePartialHeaders(msg []byte, size int) []byte {
```

Comment:
Could you provide some details about what's going here and why we need this function please ? It would be nice to have an illustration like the one we drew out yesterday.